src/NextGen/examples/MParticleCommerceHandler.ts

```diff
@@ -0,0 +1,40 @@
+import { CommerceEvent } from '@mparticle/event-models';
+import { BaseMparticleDataHandlerStrategy } from '../dataHandler/dataHandlerStrategy';
+import { BaseMPMessage } from '../core/baseMPMessage';
+import { MParticleDataRepository } from '../dataRepository';
+import BatchManager from '../batchManager';
+import { DataHandlerType } from '../dataHandler/dataHandlerType';
+
+export default class MParticleCommerceHandler extends BaseMparticleDataHandlerStrategy<
+    CommerceEvent,
+    BaseMPMessage
+> {
+    private dataRepository: MParticleDataRepository;
+    constructor(
+        dataRepository: MParticleDataRepository,
```

Comment:
I made a couple of comments about the repository being injected in the BatchManager. I think that maybe we should inject it in that layer instead of here.